lms/djangoapps/verify_student/tests/test_views.py

```diff
@@ -1307,6 +1310,80 @@ def test_submit_photos_and_change_name(self):
         # Check that the user's name was changed in the database
         self._assert_user_name(self.FULL_NAME)
 
+    def test_submit_photos_sends_confirmation_email(self):
+        self._submit_photos(
+            face_image=self.IMAGE_DATA,
+            photo_id_image=self.IMAGE_DATA
+        )
+        self._assert_confirmation_email(True)
+
+    def test_submit_photos_error_does_not_send_email(self):
+        # Error because invalid parameters, so no confirmation email
+        # should be sent.
+        self._submit_photos(expected_status_code=400)
+        self._assert_confirmation_email(False)
+
+    # Disable auto-auth since we will be intercepting POST requests
+    # to the verification service ourselves in this test.
+    @patch.dict(settings.FEATURES, {'AUTOMATIC_VERIFY_STUDENT_IDENTITY_FOR_TESTING': False})
+    @override_settings(VERIFY_STUDENT={
+        "SOFTWARE_SECURE": {
+            "API_URL": "https://verify.example.com/submit/",
+            "API_ACCESS_KEY": "dcf291b5572942f99adaab4c2090c006",
+            "API_SECRET_KEY": "c392efdcc0354c5f922dc39844ec0dc7",
+            "FACE_IMAGE_AES_KEY": "f82400259e3b4f88821cd89838758292",
+            "RSA_PUBLIC_KEY": (
+                "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDkgtz3fQdiXshy/RfOHkoHlhx/"
+                "SSPZ+nNyE9JZXtwhlzsXjnu+e9GOuJzgh4kUqo73ePIG5FxVU+mnacvufq2cu1SOx"
+                "lRYGyBK7qDf9Ym67I5gmmcNhbzdKcluAuDCPmQ4ecKpICQQldrDQ9HWDxwjbbcqpVB"
+                "PYWkE1KrtypGThmcehLmabf6SPq1CTAGlXsHgUtbWCwV6mqR8yScV0nRLln0djLDm9d"
+                "L8tIVFFVpAfBaYYh2Cm5EExQZjxyfjWd8P5H+8/l0pmK2jP7Hc0wuXJemIZbsdm+DSD"
+                "FhCGY3AILGkMwr068dGRxfBtBy/U9U5W+nStvkDdMrSgQezS5+V test@example.com"
+            ),
+            "AWS_ACCESS_KEY": "c987c7efe35c403caa821f7328febfa1",
+            "AWS_SECRET_KEY": "fc595fc657c04437bb23495d8fe64881",
+            "S3_BUCKET": "test.example.com",
+        }
+    })
+    @moto.mock_s3
+    def test_submit_photos_for_reverification(self):
```

Comment:
Note to reviewers: this test case validates the fix for a critical bug.  Before the fix, accessing the URL for the reverification's photo ID image would have resulted in a 404, since the URL is for an attempt that has a different ID than the initial attempt.